integrations/gcp/gcp_core/search/searches.py

```diff
@@ -0,0 +1,186 @@
+    The format to get the most updated value of a resource's property in the port app config is:
+
+        .versioned_resources | max_by(.version).resource | <resource_property>
+
+    for example, to get the object retention load of a bucket the currect way of getting that is:
+
+        .versioned_resources | max_by(.version).resource | .objectRetention.mode
+
+    """
+    logger.info(f"Searching all {asset_type}'s in project {project_name}")
+    async with AssetServiceAsyncClient() as async_assets_client:
+        search_all_resources_request = {
+            "scope": project_name,
+            "asset_types": [asset_type],
+            "read_mask": "*",
+        }
+        if asset_name:
+            search_all_resources_request["query"] = f"name={asset_name}"
+        try:
+            paginated_responses: pagers.SearchAllResourcesAsyncPager = (
+                await async_assets_client.search_all_resources(
+                    search_all_resources_request
+                )
+            )
+            async for paginated_response in paginated_responses.pages:
+                resources = parse_protobuf_messages(paginated_response.results)
+                if resources:
+                    logger.info(f"Generating {len(resources)} {asset_type}'s")
+                    yield resources
+        except PermissionDenied as e:
+            logger.error(f"Couldn't access the API to get kind {asset_type}: {str(e)}")
+
+
+async def list_all_topics_per_project(
+    project: dict[str, Any],
+) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    This lists all Topics under a certain project.
+    The Topics are handled specifically due to lacks of data in the asset itselfwithin the asset inventory - e.g. some properties missing.
+    The listing is being done via the PublisherAsyncClient, ignoring state in assets inventory
+    """
+    project_name = project["name"]
+    logger.info(
+        f"Searching all {AssetTypesWithSpecialHandling.TOPIC}'s in project {project_name}"
+    )
+    async with PublisherAsyncClient() as async_publisher_client:
+        try:
+            list_topics_pagers = await async_publisher_client.list_topics(
+                project=project_name
+            )
+            async for paginated_response in list_topics_pagers.pages:
+                topics = parse_protobuf_messages(paginated_response.topics)
+                if topics:
+                    logger.info(
+                        f"Generating {len(topics)} {AssetTypesWithSpecialHandling.TOPIC}'s"
+                    )
+                    for topic in topics:
+                        topic[EXTRA_PROJECT_FIELD] = project
+                    yield topics
+        except PermissionDenied:
+            logger.info(
+                f"Service account doesn't have permissions to list topics from project {project_name}"
+            )
+        except NotFound:
+            logger.debug(f"Project {project_name} is deleted")
```

Comment:
Shouldnt we fail for this?

Reply:
Nope
If the use gave us org level permissions there might be some deleted projects which still exist in the inventory, ne need to fail the entire integration for this